### ISSUE TITLE
phpmd ignore SuppressWarnings(PHPMD.UnusedFormalParameter) inside class

### ISSUE THREAD

**Issue by SailorMax** [phpmd/phpmd#505]
``` php
class testClass
{
	/** @SuppressWarnings(PHPMD.UnusedFormalParameter) */
	public function Method1($a)
	{
	}

	public function Method2($a)	/** @SuppressWarnings(PHPMD.UnusedFormalParameter) */
	{
	}
}

function Func1($a)	/** @SuppressWarnings(PHPMD.UnusedFormalParameter) */
{
}
```

Expected: no warnings

Actual: `Avoid unused parameters such as '$a'.` for Method2()

**Comment by kylekatarnls** [comment]
It's have been labelled as a bug, but I don't think it is. I don't find anywhere in the documentation where we mention `@SuppressWarnings()` can be put after the code/method where it should supress the warnings.

And I don't think we should support this feature.

If `@SuppressWarnings()` can be put before or after the code that violate the rule, then in many case, it will be unclear what part of the code it applies to. It would also "hide" the warning mute deeper in the code, when I think the PHPDoc of methods is the right place to mention the suppressed warnings and the reason:

```php
/**
 * @SuppressWarnings(PHPMD.UnusedFormalParameter) $a will be used in the next version to specify...
 */
function Func1($a)
{
}
```

`@SuppressWarnings` should be part of the method/function documentation.

**Comment by SailorMax** [comment]
Then why we have not warning for function Func1()?
Does functions and methods has different rules?

**Comment by kylekatarnls** [comment]
Func1() warning should not be suppressed as the documentation does not mention this syntax: https://phpmd.org/documentation/suppress-warnings.html

We will discuss about supporting the suppression at the end of the line, but you should better use the method/function doc block instead. For example UnusedLocalVariable only works with doc blocks.

So I guess if we change it in the future, we'll more likely align UnusedFormalParameter to other rules.
